addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParam.java

```diff
@@ -170,6 +171,17 @@ public ArgsTypeOption getArgsType() {
         return argsType;
     }
 
+    // For generating an API action.
+    public void setArgsType(String argsType) {
+        try {
+            setArgsType(ArgsTypeOption.valueOf(argsType.toUpperCase(Locale.ROOT)));
+        } catch (IllegalArgumentException e) {
+            LOG.error('"' + argsType + "\" is not a valid Arguments Specification Type.");
```

Comment:
Core should already log the problem (in any case this should be less than error, since it's bad input not an issue with the code).